testapp/src/main/java/com/rakuten/tech/mobile/testapp/analytics/rat_wrapper/RATActivity.kt

```
@@ -0,0 +1,118 @@
+        }
+    }
+
+    /** Will send Appear event when open a activity. */
+    private fun sendAnalyticsDefault() {
+        prepareScreenAppearEventToSend()
+        ratEvent?.let {
+            DemoAppAnalytics.init(AppSettings.instance.projectId).sendAnalytics(it)
+        }
+    }
+
+    /** Prepare default event if custom RAT event is not set. */
+    override fun prepareMenuItemClickEventToSend() {
+        if (ratEventMenuItem == null)
+            ratEventMenuItem = RATEvent(
+                event = EventType.CLICK,
+                action = ActionType.OPEN,
+                label = menu_item_label,
+                screen_name = getScreenName()
+            )
+    }
+
+    /** Prepare default event if custom RAT event is not set. */
+    override fun prepareScreenAppearEventToSend() {
+        if (ratEvent == null)
+            ratEvent = RATEvent(
+                event = EventType.APPEAR,
+                action = ActionType.OPEN,
+                label = "",
+                screen_name = getScreenName()
+            )
+    }
+
+    /** Can send custom RAT event if set. */
+    override fun setCustomRatEventForMenuItem(ratEvent: RATEvent) {
+        this.ratEventMenuItem = ratEvent
+    }
+
+    /** Can send custom RAT event if set. */
+    override fun setCustomRatEventForDefault(ratEvent: RATEvent) {
+        this.ratEvent = ratEvent
+    }
+
+    /** Can clear custom RAT event if set. */
+    override fun clearCustomRatEvent() {
+        this.ratEvent = null
+    }
+
+    /** Can clear custom RAT event if set. */
+    override fun clearCustomRatEventforMenuItem() {
+        this.ratEventMenuItem = null
+    }
+
+    /**
+     * Get the screen name of the activity
+     * @return the name of the screen to be reported to RAT for a screen view event
+     * @throws UnsetScreenException if screen view is not set by overriding this method
+     */
+    @Throws(UnsetScreenException::class)
```

I am not sure if we need to throw any exception for the screen name, if we don't set, it still can send blank name 🤔